akd/src/errors.rs

```diff
@@ -51,112 +51,78 @@ impl From<AzksError> for AkdError {
 
 impl From<StorageError> for HistoryTreeNodeError {
     fn from(error: StorageError) -> Self {
-        Self::StorageError(error)
+        Self::Storage(error)
+    }
+}
+
+impl std::fmt::Display for AkdError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> Result<(), std::fmt::Error> {
+        writeln!(f, "AkdError: {:?}", self)
     }
 }
 
 /// Errors thown by HistoryTreeNodes
 #[derive(Debug)]
 pub enum HistoryTreeNodeError {
-    /// Tried to set a child and the direction given was none.
-    NoDirectionInSettingChild(u64, u64),
-    /// Direction is unexpectedly None
-    DirectionIsNone,
+    /// No direction provided for the node.
+    /// Second parameter is the label of the child attempted to be set
+    /// -- if there is one, otherwise it is None.
+    NoDirection(u64, Option<u64>),
     /// The node didn't have a child in the given epoch
-    NoChildInTreeAtEpoch(u64, usize),
-    /// The node had no children at the given epoch
-    NoChildrenInTreeAtEpoch(u64),
-    /// The hash was being updated for an invalid epoch
-    InvalidEpochForUpdatingHash(u64),
-    /// Tried to update the parent of the root, which should not be done
-    TriedToUpdateParentOfRoot,
+    NoChildAtEpoch(u64, usize),
     /// The next epoch of this node's parent was invalid
     ParentNextEpochInvalid(u64),
     /// The hash of a parent was attempted to be updated, without setting the calling node as a child.
-    HashUpdateOnlyAllowedAfterNodeInsertion,
-    /// The children of a leaf are always dummy and should not be hashed
-    TriedToHashLeafChildren,
-    /// The list of epochs for a given node was empty
-    NodeCreatedWithoutEpochs(u64),
-    /// The label of a leaf node was shorter than that of an interior node.
-    LeafNodeLabelLenLessThanInterior(NodeLabel),
-    /// Error compressing the Merkle trie
-    CompressionError(NodeLabel),
-    /// Tried to access something about the node at an epoch that didn't exist.
-    NodeDidNotExistAtEp(NodeLabel, u64),
+    HashUpdateOrderInconsistent,
+    /// The node did not exist at epoch
+    NonexistentAtEpoch(NodeLabel, u64),
     /// The state of a node did not exist at a given epoch
-    NodeDidNotHaveExistingStateAtEp(NodeLabel, u64),
-    /// Error propogation
-    StorageError(StorageError),
-    /// Error propogation
+    NoStateAtEpoch(NodeLabel, u64),
+    /// Error propagation
+    Storage(StorageError),
+    /// Error propagation
     SerializationError,
```

Comment:
Is this error ever called? Storage errors also have a serialization error, so I'm curious if this one is utilized anywhere

Reply:
The other `SerializationError` was not used; I removed it.

I guess the propagation comment is unnecessary now...